src/react-components/room/PeopleSidebarContainer.js

```diff
@@ -0,0 +1,142 @@
+import React, { useCallback, useEffect, useState } from "react";
+import PropTypes from "prop-types";
+import { PeopleSidebar } from "./PeopleSidebar";
+import { getMicrophonePresences } from "../../utils/microphone-presence";
+import ProfileEntryPanel from "../profile-entry-panel";
+import ClientInfoDialog from "../client-info-dialog";
+
+export function userFromPresence(sessionId, presence, micPresences, mySessionId) {
+  const meta = presence.metas[presence.metas.length - 1];
+  const micPresence = micPresences.get(sessionId);
+  return { id: sessionId, isMe: mySessionId === sessionId, micPresence, ...meta };
+}
+
+function usePeopleList(presences, mySessionId, micUpdateFrequency = 500) {
+  const [people, setPeople] = useState([]);
+
+  useEffect(
+    () => {
+      let timeout;
+
+      function updateMicrophoneState() {
+        const micPresences = getMicrophonePresences();
+
+        setPeople(
+          Object.entries(presences).map(([id, presence]) => {
+            return userFromPresence(id, presence, micPresences, mySessionId);
+          })
+        );
+
+        timeout = setTimeout(updateMicrophoneState, micUpdateFrequency);
+      }
+
+      updateMicrophoneState();
+
+      return () => {
+        clearTimeout(timeout);
+      };
+    },
+    [presences, micUpdateFrequency, setPeople, mySessionId]
+  );
+
+  return people;
+}
+
+function PeopleListContainer({ hubChannel, presences, mySessionId, onSelectPerson, onClose }) {
+  const people = usePeopleList(presences, mySessionId);
+
+  const onMuteAll = useCallback(
+    () => {
+      for (const person of people) {
+        if (person.presence === "room" && person.permissions && !person.permissions.mute_users) {
+          hubChannel.mute(person.id);
```

Comment:
Not sure if I'm doing something wrong but I don't seem to be reaching the line. (Mute all doesn't seem to be working)

Reply:
Can you expand on this? Were you logged in? Was there another user in the room? Were they not muted?

Reply:
Took me a bit to figure out -- the case I was running into was that I was logged in as the same user twice, so the check for `person.permissions.mute_users` prevented the mute from going thru. Is this behavior correct? (Is "mute all" not supposed to apply to admins / owners?)

Reply:
It doesn't currently check for that: https://github.com/mozilla/hubs/blob/master/src/react-components/presence-list.js#L195